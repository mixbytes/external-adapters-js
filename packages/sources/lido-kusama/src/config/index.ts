import { Requester, util } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

export const NAME = 'LIDO_KUSAMA'
export const KSM_AGGREGATOR_PROXY = '0x6e0513145FCE707Cd743528DB7C1cAB537DE9d1B'
export const STKSM_RATE_PROVIDER = '0x77D4b212770A7cA26ee70b1E0f27fC36da191c53'
export const DEFAULT_ENDPOINT = 'stksm'
export const OUTPUT_DECIMALS = 8

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  config.api.baseURL ||= util.getEnv('MOONRIVER_RPC_URL') || ''
  config.defaultEndpoint = DEFAULT_ENDPOINT
  return config
}
