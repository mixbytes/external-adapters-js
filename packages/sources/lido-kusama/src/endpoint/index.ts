export * as stksm from './stksm'
