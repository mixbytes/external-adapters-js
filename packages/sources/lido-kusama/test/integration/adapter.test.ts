import { AdapterRequest } from '@chainlink/types'
import request, { SuperTest, Test } from 'supertest'
import process from 'process'
import nock from 'nock'
import { server as startServer } from '../../src'
import { mockRelaychainSuccess, mockParachainSuccess } from './fixtures'
import { AddressInfo } from 'net'

let oldEnv: NodeJS.ProcessEnv

beforeAll(() => {
  oldEnv = JSON.parse(JSON.stringify(process.env))
  process.env.KUSAMA_RPC_URL = process.env.KUSAMA_RPC_URL || 'https://test-rpc-url-relay'
  process.env.MOONRIVER_RPC_URL = process.env.MOONRIVER_RPC_URL || 'https://test-rpc-url-para'
  if (process.env.RECORD) nock.recorder.rec()
})

afterAll(() => {
  process.env = oldEnv
  if (process.env.RECORD) nock.recorder.play()

  nock.restore()
  nock.cleanAll()
  nock.enableNetConnect()
})

describe('execute', () => {
  let fastify: FastifyInstance
  let req: SuperTest<Test>

  beforeAll(async () => {
    fastify = await startServer()
    req = request(`localhost:${(fastify.server.address() as AddressInfo).port}`)
  })

  afterAll((done) => {
    fastify.close(done)
  })

  describe('stksm endpoint', () => {
    const data: AdapterRequest = {
      id: '1',
      data: {},
    }

    it('relaychain should return success', async () => {
      mockRelaychainSuccess()

      const response = await req
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    }, 60000)

    it('parachain should return success', async () => {
      mockParachainSuccess()

      const response = await req
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    }, 60000)
  })
})
