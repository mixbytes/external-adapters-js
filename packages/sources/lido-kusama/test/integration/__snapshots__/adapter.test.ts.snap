// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute stksm endpoint should return success 1`] = `
Object {
  "data": Object {
    "result": 6684919487,
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": 6684919487,
  "statusCode": 200,
}
`;